R/zchunk_LB123.Electricity_china.R

```diff
@@ -0,0 +1,262 @@
+#' module_gcam.china_LB123.Electricity
+#'
+#' Electricity sector inputs and outputs, and electricity ownuse.
+#'
+#' @param command API command to execute
+#' @param ... other optional parameters, depending on command
+#' @return Depends on \code{command}: either a vector of required inputs,
+#' a vector of output names, or (if \code{command} is "MAKE") all
+#' the generated outputs: \code{L123.in_EJ_province_elec_F}, \code{L123.out_EJ_province_elec_F}, \code{L123.in_EJ_province_ownuse_elec}, \code{L123.out_EJ_province_ownuse_elec}. The corresponding file in the
+#' original data system was \code{LB123.Electricity.R} (gcam-china level1).
+#' @details Compute electricity sector inputs and outputs, and electricity ownuse.
+#' @importFrom assertthat assert_that
+#' @importFrom dplyr filter mutate select
+#' @importFrom tidyr gather spread
+#' @author Liu August 2018
+module_gcam.china_LB123.Electricity <- function(command, ...) {
+  if(command == driver.DECLARE_INPUTS) {
+    return(c(FILE = "gcam-china/CPSY_GWh_province_F_elec_out",
+             FILE = "gcam-china/province_names_mappings",
+             "L123.in_EJ_R_elec_F_Yh",
+             "L123.out_EJ_R_elec_F_Yh",
+             "L126.in_EJ_R_elecownuse_F_Yh",
+             "L126.out_EJ_R_elecownuse_F_Yh",
+             "L101.inNBS_Mtce_province_S_F",
+             "L132.out_EJ_province_indchp_F"))
+  } else if(command == driver.DECLARE_OUTPUTS) {
+    return(c("L123.in_EJ_province_elec_F",
+             "L123.out_EJ_province_elec_F",
+             "L123.in_EJ_province_ownuse_elec",
+             "L123.out_EJ_province_ownuse_elec"))
+  } else if(command == driver.MAKE) {
+
+    # =============================================================================
+    other <- row <- oil <- natural.gas <- coal<- coal.adj <- wind <- wind.adj<- nuclear<- nuc.adj <-
+      solar <- solar.adj <- biomass <- fuel <- value <- province  <- sector <- year <- copy_rl <- copy_g <-
+      pct <- each <-Sum <- value.x <- value.y <- sector.x <- sector.y <- fuel.x <- fuel.y <- NULL     # silence package check.
+    all_data <- list(...)[[1]]
+    HISTORICAL_YEARS_temp <- c(HISTORICAL_YEARS, 2011, 2012)
```

Comment:
Isn't it the same as `gcamchina.ELEC_HISTORICAL_YEARS` defined in constant? Can we use that instead?